### DISCUSSION TITLE
atomWithObservable: getObservable cannot be async

### DISCUSSION THREAD

**Discussion by jeanfortheweb** [pmndrs/jotai#2043] [Bug report]
Hello,

since the latest updates of jotai all atoms are forced to be async if atoms they work with are async itself. 
However, the getObservable parameter of atomWithObservable does not support async operations. 

I tested it locally and made the internal atoms of atomWithObservable async and also allowed the getObservable to be async and it seems to work just fine.

Any plans to support this officially?

I tried to find such a discussion but failed to find one.

Thanks in advance!

**Comment by dai-shi** [comment]
We plan to finalize `unstable_unwrap` soon, but I would like to try it at this point to see if it solves your case and give feedback before finalizing.

On the other hand, in general, observables supersedes promises. So, I wonder if you change `getObservable` to return an observable from a promise.

**Reply by jeanfortheweb** [reply]
Thanks for your reply @dai-shi, unwrap seems to work just fine for the moment!

**Comment by dai-shi** [comment]
> On the other hand, in general, observables supersedes promises. So, I wonder if you change getObservable to return an observable from a promise.

Something like this should do:
```js
atomWithObservable((get) => {
  const promise = get(promiseAtom);
  return defer(() => promise).pipe(switchMap(() => from(['a', 'b', 'c'])));
})
```
